src/applications/claims-status/actions/index.jsx

```diff
@@ -147,16 +148,55 @@ export function fetchClaimsSuccess(response) {
   };
 }
 
-export function getClaimsV2() {
+const POLLING_INTERVAL = 1000;
+
+export function pollClaimsStatus({
+  onError,
+  onSuccess,
+  pollingInterval,
+  request = apiRequest,
+  shouldFail,
+  shouldSucceed
+}) {
+  return request(
+    '/evss_claims_async',
+    null,
+    response => {
+      if (shouldSucceed(response)) {
+        onSuccess(response);
+        return;
+      }
+
+      if (shouldFail(response)) {
+        onError(response);
+        return;
+      }
+
+      setTimeout(
+        () => pollClaimsStatus({ onError, onSuccess, pollingInterval, request, shouldFail, shouldSucceed }),
+        pollingInterval
+      );
+    },
+    error => onError(error)
+  );
+}
+
+export function getSyncStatus(claimsAsyncResponse) {
+  return get('meta.syncStatus', claimsAsyncResponse, null);
+}
+
+export function getClaimsV2(poll = pollClaimsStatus) {
+
   return (dispatch) => {
     dispatch({ type: FETCH_CLAIMS_PENDING });
-    return apiRequest(
-      '/evss_claims',
-      null,
-      (response) => dispatch(fetchClaimsSuccess(response)),
-      // TO-DO: parse out errors, log in Sentry
-      () => dispatch({ type: FETCH_CLAIMS_ERROR })
-    );
+
+    poll({
+      onError: () => dispatch({ type: FETCH_CLAIMS_ERROR }),
+      onSuccess: response => dispatch(fetchClaimsSuccess(response)),
+      pollingInterval: window.VetsGov.pollTimeout || POLLING_INTERVAL,
```

Comment:
Do we anticipate using this in a setting where we would want to be able to configure the pollingInterval, rather than accessing it as a variable in the outer scope?

Reply:
Nope- I would rather use the hard coded value since it's not reused. I'll update this.